tests/baselines/reference/controlFlowArrayErrors.errors.txt

```diff
@@ -77,8 +76,6 @@ controlFlowArrayErrors.ts(63,9): error TS7005: Variable 'x' implicitly has an 'a
         }
         x;           // boolean[] | (string | number)[]
         x.push(99);  // Error

```

Comment:
The "non-evolving-array-ness" is lost as `boolean[] | (string | number)[]` becomes `(boolean | string | number)[]` in the Array.push call.